backend/benefit/applications/api/v1/serializers.py

```diff
@@ -897,97 +916,41 @@ def _validate_date_range(self, start_date, end_date, benefit_type):
-        ).order_by("-start_date")
-        if self.instance:
-            previously_granted_benefits = previously_granted_benefits.exclude(
-                pk=self.instance.pk
-            )
-
-        self._validate_no_benefit_overlap(
-            start_date, end_date, previously_granted_benefits
-        )
-
-        most_recent_benefit = previously_granted_benefits.first()
-        if (
-            not most_recent_benefit
-            or most_recent_benefit.end_date
-            < start_date - relativedelta(months=self.BENEFIT_WAITING_PERIOD_MONTHS)
-        ):
-            # at least BENEFIT_WAITING_PERIOD_MONTHS elapsed since a Helsinki benefit was granted
-            # for this employee last time.
-            return
-
-        # scroll back previous benefits until we find a gap of BENEFIT_WAITING_PERIOD_MONTHS
-        applicable_benefits = [most_recent_benefit]
-        for old_benefit, older_benefit in pairwise(previously_granted_benefits):
-            # on the first round of loop, old_benefit == most_recent_benefit, which already is in
-            # applicable_benefits
-            if (
-                older_benefit.end_date
-                + relativedelta(months=self.BENEFIT_WAITING_PERIOD_MONTHS)
-                < old_benefit.start_date
-            ):
-                break
-            applicable_benefits.append(older_benefit)
-
-        past_months_of_benefit = sum(
-            benefit.duration_in_months for benefit in applicable_benefits
-        )
-        if (
-            past_months_of_benefit + duration_in_months(start_date, end_date)
-            > self.BENEFIT_MAX_MONTHS
-        ):
-            # granting this benefit would exceed the limit
-            raise serializers.ValidationError(
-                {
-                    "start_date": _(
-                        "Benefit can not be granted before 24-month waiting period expires"
-                    )
-                }
-            )
-
-    def _validate_no_benefit_overlap(
-        self, start_date, end_date, previously_granted_benefits
-    ):
-        for benefit in previously_granted_benefits:
-            if date_range_overlap(
-                start_date, end_date, benefit.start_date, benefit.end_date
-            ):
-                raise serializers.ValidationError(
-                    {
-                        "start_date": _(
-                            "There's already an accepted application with overlapping date range for this employee"
-                        )
-                    }
-                )
+        if validation_errors := get_past_benefit_info(
+            self.instance,
+            company,
+            social_security_number,
+            start_date,
+            end_date,
+            apprenticeship_program,
+        ).validation_errors:
+            error_text = ", ".join(str(v) for v in validation_errors)
+            raise serializers.ValidationError({"start_date": error_text})
```

Comment:
Is the validation error here always related specifically to start_date or can it be related to something else?

Reply:
Rest framework stores errors under some key. It's possible to use a generic error also, but that loses all context of the error. In the frontend, the error message structure is at the moment flattened AFAIK, so the key doesn't have effect on the display of the error.

Reply:
As discussed, in future this error will be turned into a warning, and the issue can be addressed at thata time.

Reply:
> As discussed, in future this error will be turned into a warning, and the issue can be addressed at thata time.

👌